src/arrayMethodReduce.js

```diff
@@ -4,8 +4,20 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, initialValue = 0) {
+    let i = 0;
+    let accumulator = initialValue;
+
+    if (arguments.length === 1) {
+      accumulator = this[0];
```

Comment:
instead of doing this, you can specify default value for `initialValue` and use ternary operator for `i`